### PR TITLE
Ignore pl/pgsql line numbers in regression outputs

### PR DESCRIPTION
**Pull request by hanefi**
When I run the multi_extension test on my Mac, I see a differing line number and the test fail.

```diff
diff --git a/src/test/regress/expected/multi_extension.out b/src/test/regress/expected/multi_extension.out
index 4e002550..11ff6967 100644
--- a/src/test/regress/expected/multi_extension.out
+++ b/src/test/regress/expected/multi_extension.out
@@ -437,7 +437,7 @@ BEGIN;
   ALTER EXTENSION citus UPDATE TO '9.4-1';
 ERROR:  citus local tables are introduced in Citus 9.5
 HINT:  To downgrade Citus to an older version, you should first convert each citus local table to a postgres table by executing SELECT undistribute_table("%s")
-CONTEXT:  PL/pgSQL function inline_code_block line 11 at RAISE
+CONTEXT:  PL/pgSQL function inline_code_block line 12 at RAISE
 ROLLBACK;
 -- now we can downgrade as there is no citus local table
 ALTER EXTENSION citus UPDATE TO '9.4-1';
```

### CODE REVIEW

**Comment by hanefi** [comment]
I use BSD sed instead of the GNU sed. Maybe this is why my regexes do not work on CI.

**Comment by hanefi** [comment]
@onurctirtir can you take a look?

File: src/test/regress/bin/normalize.sed [hunk 1]
```diff
@@ -193,3 +193,6 @@ s/relation with OID [0-9]+ does not exist/relation with OID XXXX does not exist/
 
 # normalize storage id of columnar tables
 s/^storage id: [0-9]+$/storage id: xxxxx/g
+
+# ignore PL/pgSQL line numbers that differ on Mac builds
+s/(CONTEXT:  PL\/pgSQL function inline_code_block line )(\d+)/\1XX/g
```
Comment (onurctirtir) [inline src/test/regress/bin/normalize.sed:198]:
```suggestion
s/(CONTEXT:  PL\/pgSQL function inline_code_block line )([0-9]+)/\1XX/g
```

This works on my machine

**Review by onurctirtir — COMMENTED** [review]
I left a suggestion to fix ci failures, however, I think we should first understand why such line numbers differ across platforms, can you check that ?

Plus, it might be better if you can verify this with other Mac users in the team

File: src/test/regress/bin/normalize.sed (hunk 1 above)
Comment (hanefi) [reply src/test/regress/bin/normalize.sed]: I made the change you suggested, and ran `./ci/normalize_expected.sh` and update all affected test outputs. Some tests still fail

Comment (onurctirtir) [reply src/test/regress/bin/normalize.sed]: should we use `make reindent` instead ? (not sure if it helps)

**Comment by hanefi** [comment]
I compared built migration file contents on my Mac, and Onur's Ubuntu WSL. It appears that my machine does not strip empty lines from the migration scripts and this creates a discrepancy in line numbers in pl/pgsql functions

```diff
sh$ diff -Naur hanefi-citus--9.5-1--9.4-1.sql onur-citus--9.5-1--9.4-1.sql
--- hanefi-citus--9.5-1--9.4-1.sql    2020-12-15 11:06:22.000000000 +0300
+++ onur-citus--9.5-1--9.4-1.sql    2020-12-15 13:43:37.000000000 +0300
@@ -1,12 +1,5 @@
-
-
-
-
 -- citus--9.5-1--9.4-1
-
 SET search_path = 'pg_catalog';
-
-
 CREATE OR REPLACE FUNCTION pg_catalog.citus_drop_trigger()
     RETURNS event_trigger
     LANGUAGE plpgsql
@@ -22,7 +15,6 @@
     SELECT array_agg(object_identity) INTO sequence_names
     FROM pg_event_trigger_dropped_objects()
     WHERE object_type = 'sequence';
-
     FOR v_obj IN SELECT * FROM pg_event_trigger_dropped_objects()
                  WHERE object_type IN ('table', 'foreign table')
     LOOP
@@ -33,11 +25,9 @@
         PERFORM master_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name);
         PERFORM master_remove_partition_metadata(v_obj.objid, v_obj.schema_name, v_obj.object_name);
     END LOOP;
-
     IF cardinality(sequence_names) > 0 THEN
         PERFORM master_drop_sequences(sequence_names);
     END IF;
-
     -- remove entries from citus.pg_dist_object for all dropped root (objsubid = 0) objects
     FOR v_obj IN SELECT * FROM pg_event_trigger_dropped_objects()
     LOOP
@@ -47,10 +37,6 @@
 $cdbdt$;
 COMMENT ON FUNCTION pg_catalog.citus_drop_trigger()
     IS 'perform checks and actions at the end of DROP actions';
-
-
-
-
 -- Check if user has any citus local tables.
 -- If not, DROP create_citus_local_table UDF and continue safely.
 -- Otherwise, raise an exception to stop the downgrade process.
@@ -60,7 +46,6 @@
 BEGIN
     SELECT COUNT(*) INTO citus_local_table_count
     FROM pg_dist_partition WHERE repmodel != 't' AND partmethod = 'n';
-
     IF citus_local_table_count = 0 THEN
         -- no citus local tables exist, can safely downgrade
         DROP FUNCTION create_citus_local_table(table_name regclass);
@@ -72,71 +57,57 @@
     END IF;
 END;
 $$ LANGUAGE plpgsql;
-
 DROP FUNCTION worker_record_sequence_dependency(regclass, regclass, name);
-
---  task_tracker_* functions
-
+-- task_tracker_* functions
 CREATE FUNCTION task_tracker_assign_task(bigint, integer, text)
     RETURNS void
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$task_tracker_assign_task$$;
 COMMENT ON FUNCTION task_tracker_assign_task(bigint, integer, text)
     IS 'assign a task to execute';
-
 CREATE FUNCTION task_tracker_task_status(bigint, integer)
     RETURNS integer
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$task_tracker_task_status$$;
 COMMENT ON FUNCTION task_tracker_task_status(bigint, integer)
     IS 'check an assigned task''s execution status';
-
 CREATE FUNCTION task_tracker_cleanup_job(bigint)
     RETURNS void
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$task_tracker_cleanup_job$$;
 COMMENT ON FUNCTION task_tracker_cleanup_job(bigint)
     IS 'clean up all tasks associated with a job';
-
 CREATE FUNCTION worker_merge_files_and_run_query(bigint, integer, text, text)
     RETURNS void
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$worker_merge_files_and_run_query$$;
 COMMENT ON FUNCTION worker_merge_files_and_run_query(bigint, integer, text, text)
     IS 'merge files and run a reduce query on merged files';
-
 CREATE FUNCTION worker_execute_sql_task(jobid bigint, taskid integer, query text, binary bool)
     RETURNS bigint
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$worker_execute_sql_task$$;
 COMMENT ON FUNCTION worker_execute_sql_task(bigint, integer, text, bool)
     IS 'execute a query and write the results to a task file';
-
 CREATE FUNCTION task_tracker_conninfo_cache_invalidate()
     RETURNS trigger
     LANGUAGE C
     AS 'citus', $$task_tracker_conninfo_cache_invalidate$$;
 COMMENT ON FUNCTION task_tracker_conninfo_cache_invalidate()
     IS 'invalidate task-tracker conninfo cache';
-
 CREATE TRIGGER dist_poolinfo_task_tracker_cache_invalidate
     AFTER INSERT OR UPDATE OR DELETE OR TRUNCATE
     ON pg_catalog.pg_dist_poolinfo
     FOR EACH STATEMENT EXECUTE PROCEDURE task_tracker_conninfo_cache_invalidate();
-
 CREATE TRIGGER dist_authinfo_task_tracker_cache_invalidate
     AFTER INSERT OR UPDATE OR DELETE OR TRUNCATE
     ON pg_catalog.pg_dist_authinfo
     FOR EACH STATEMENT EXECUTE PROCEDURE task_tracker_conninfo_cache_invalidate();
-
 CREATE FUNCTION master_drop_sequences(sequence_names text[])
     RETURNS void
     LANGUAGE C STRICT
     AS 'MODULE_PATHNAME', $$master_drop_sequences$$;
 COMMENT ON FUNCTION master_drop_sequences(text[])
     IS 'drop specified sequences from the cluster';
-
 RESET search_path;
-
 DROP FUNCTION pg_catalog.undistribute_table(table_name regclass);
-
```

**Comment by hanefi** [comment]
@thanodnl can you verify that this change works on a Mac as well?

GNU sed is slightly different than BSD one. It's enough to see that `./ci/normalize_expected.sh` does not change any files I guess

**Comment by thanodnl** [comment]
Running `./ci/normalize_expected.sh` does not change any files.
However, running the `check-multi` schedule causes the following failure:

```
+++ /Users/nilsdijk/Projects/citus/src/test/regress/results/alter_table_set_access_method.out.modified	2021-05-26 12:44:44.000000000 +0200
@@ -184,21 +184,21 @@
 
 SELECT * FROM partitioned_table_6_10 ORDER BY 1, 2;
  id | a 
 ----+---
   7 | 2
 (1 row)
 
 -- try to compress partitions with an integer partition column
 CALL alter_old_partitions_set_access_method('partitioned_table', '2021-01-01', 'columnar');
 ERROR:  partition column of partitioned_table cannot be cast to a timestamptz
-CONTEXT:  PL/pgSQL function alter_old_partitions_set_access_method(regclass,timestamp with time zone,name) line 13 at RAISE
+CONTEXT:  PL/pgSQL function alter_old_partitions_set_access_method(regclass,timestamp with time zone,name) line 14 at RAISE
 CREATE TABLE time_partitioned (event_time timestamp, event int) partition by range (event_time);
 SELECT create_distributed_table('time_partitioned', 'event_time');
  create_distributed_table 
 --------------------------
  
 (1 row)
 
 CREATE TABLE time_partitioned_d00 PARTITION OF time_partitioned FOR VALUES FROM ('2000-01-01') TO ('2009-12-31');
 CREATE TABLE time_partitioned_d10 PARTITION OF time_partitioned FOR VALUES FROM ('2010-01-01') TO ('2019-12-31');
 CREATE TABLE time_partitioned_d20 PARTITION OF time_partitioned FOR VALUES FROM ('2020-01-01') TO ('2029-12-31');
```

**Review by JelteF — APPROVED** [review]

**Pull request merged**